src/main/java/apoc/hashing/Fingerprinting.java

```diff
@@ -0,0 +1,198 @@
+package apoc.hashing;
+
+import org.neo4j.graphdb.*;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.UserFunction;
+
+import javax.xml.bind.DatatypeConverter;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class Fingerprinting {
+
+    public static final String DIGEST_ALGORITHM = "MD5";
+
+    @Context
+    public GraphDatabaseService db;
+
+    @UserFunction
+    @Description("calculate a checksum (md5) over a node or a relationship. This deals gracefully with array properties. Two identical entities do share the same hash.")
+    public String fingerprintNodeOrRel(@Name("node_relation") Object thing, @Name(value = "propertyExcludes", defaultValue = "") List<String> excludedPropertyKeys) {
+
+        if (thing instanceof Node) {
+            return fingerprintNode((Node) thing, excludedPropertyKeys);
+        } else if (thing instanceof Relationship) {
+                return fingerprintRelationship((Relationship) thing, excludedPropertyKeys);
+        } else {
+            throw new UnsupportedOperationException("cannot fingerprint a " + thing.getClass());
+        }
+
+    }
+
+    @UserFunction
```

Comment:
perhaps this is better as a procedure so it could return more columns on number of nodes, rels, properties observed ?

Reply:
hmm. I'm not sure if these additional fields do make sense. At the end I just want a hashsum to compare a graph with another one.